docs/en/user_guides/inference.md

```diff
@@ -1,32 +1,82 @@
+2. [**`ImageClassificationInferencer`**](mmpretrain.apis.ImageClassificationInferencer): inference on given images.
+3. [**`FeatureExtractor`**](mmpretrain.apis.FeatureExtractor): extract features from the image files directly.
+4. [**`ImageRetrievalInferencer`**](mmpretrain.apis.ImageRetrievalInferencer): retrieve images from a folder.
 
-## Inference on a given image
+For more details about the pre-trained models in MMPretrain, you can refer to [Model Zoo](../modelzoo_statistics.md).
+
+## List models and Get model
+
+list all the models in MMPreTrain.
+
+```
+>>> from mmpretrain import list_models
+>>> list_models()
+['barlowtwins_resnet50_8xb256-coslr-300e_in1k',
+ 'beit-base-p16_beit-in21k-pre_3rdparty_in1k',
+ .................]
+```
 
-MMPretrain provides high-level Python APIs for inference on a given image:
+`list_models` supports fuzzy matching, you can use **\*** to match any character.
 
-- [`get_model`](mmpretrain.apis.get_model): Get a model with the model name.
-- [`inference_model`](mmpretrain.apis.inference_model): Inference on a given image
+```
+>>> from mmpretrain import list_models
+>>> list_models("*convnext-b*21k")
+['convnext-base_3rdparty_in21k',
+ 'convnext-base_in21k-pre-3rdparty_in1k-384px',
+ 'convnext-base_in21k-pre_3rdparty_in1k']
+```
 
-Here is an example of building the model and inference on a given image by using ImageNet-1k pre-trained checkpoint.
+you can use `get_model` get the model.
 
-```{note}
-You can use `wget https://github.com/open-mmlab/mmpretrain/raw/main/demo/demo.JPEG` to download the example image or use your own image.
 ```
+>>> from mmpretrain import get_model
 
-```python
-from mmpretrain import get_model, inference_model
+# get your owner pretrained model
+>>> your_model = get_model("CONFIG_PATH", pretrained="CKPT_PATH")
 
-img_path = 'demo.JPEG'   # you can specify your own picture path
+# model without pre-trained weight
+>>> model = get_model("convnext-base_in21k-pre_3rdparty_in1k")
 
-# build the model from a config file and a checkpoint file
-model = get_model('resnet50_8xb32_in1k', pretrained=True, device="cpu")  # device can be 'cuda:0'
-# test a single image
-result = inference_model(model, img_path)
+# model with weight in MMPreTrain
+>>> model = get_model("convnext-base_in21k-pre_3rdparty_in1k", pretrained=True)
+
+# model with weight in local
+>>> model = get_model("convnext-base_in21k-pre_3rdparty_in1k", pretrained="your_local_checkpoint_path")
+
+# you can also do some modification, like modify the num_classes in head.
+>>> model = get_model("convnext-base_in21k-pre_3rdparty_in1k", head=dict(num_classes=10))
+
+# you can get model without neck, head, and output from stage 1, 2, 3 in backbone
+>>> model_headless = get_model("resnet18_8xb32_in1k", head=None, neck=None, backbone=dict(out_indices=(1, 2, 3)))
+```
+
+Then you can do the forward:
+
+```
+>>> import torch
+>>> x = torch.rand((1, 3, 224, 224))
+>>> y = model(x)
```

Comment:
better to indicate which model is used here

Reply:
done.